### PR TITLE
Fix stStaticCall EF tests

### LINKED ISSUE #439: Fix stStaticCall EF tests

### PR DESCRIPTION
**Pull request by belfortep**
Closes #439

## Summary
Fix the `stStaticCall` test from the EF by fixing a segmentation fault when using a `m_size` parameter bigger than needed for the `modexp` precompile

## Changes
- Deleted some tests from the ignored group as they already passed
- Added tests for the `modexp` precompile

### CODE REVIEW

File: src/precompiles.rs [hunk 1]
```diff
@@ -574,23 +580,30 @@ pub fn execute_precompile(
...
+        let e_data_size = U256::from(e_size);
+        let m_data_size = U256::from(m_size);
+        let e_size = e_size as usize;
+        let m_size = m_size as usize;
+
+        let mut calldata = vec![0_u8; calldata_size];
+        let calldata_slice = calldata.as_mut_slice();
+        b_data_size.to_big_endian(&mut calldata_slice[..32]);
+        e_data_size.to_big_endian(&mut calldata_slice[32..64]);
+        m_data_size.to_big_endian(&mut calldata_slice[64..96]);
```
Comment (juanimedone) [inline src/precompiles.rs:595]:
```suggestion
        b_data_size.to_big_endian(&mut calldata_slice[..BSIZE_END]);
        e_data_size.to_big_endian(&mut calldata_slice[BSIZE_END..ESIZE_END]);
        m_data_size.to_big_endian(&mut calldata_slice[ESIZE_END..MSIZE_END]);
```

File: src/precompiles.rs [hunk 2]
```diff
@@ -630,6 +630,70 @@ mod tests {
...
+        let calldata = calldata_for_modexp(1, 1, 1, 8, 9, 0);
+        let gas_limit = 100_000_000;
+        let mut consumed_gas = 0;
+
+        let (return_code, return_data) =
+            execute_precompile(callee_address, calldata, gas_limit, &mut consumed_gas);
+
+        assert_eq!(return_code, SUCCESS_RETURN_CODE);
+        assert_eq!(return_data, Bytes::from(0_u8.to_be_bytes().to_vec()));
+        assert_eq!(consumed_gas, 200);
```
Comment (juanimedone) [inline src/precompiles.rs:660]:
```suggestion
        assert_eq!(consumed_gas, MIN_MODEXP_COST);
```

**Review by juanimedone — APPROVED** [review]

File: src/precompiles.rs [hunk 3]
```diff
@@ -168,7 +168,13 @@ pub fn modexp(
         b.modpow(&e, &m)
     };
 
-    let output = &result.to_bytes_be()[..m_size];
+    let mut bytes = result.to_bytes_be();
+    if bytes.len() < m_size {
+        bytes.resize(m_size, 0);
+        bytes.reverse();
```
Comment (TomasArrachea) [inline src/precompiles.rs:174]: Why is the reverse needed?

Comment (belfortep) [reply src/precompiles.rs:174]: It's the fault of the `resize`, if we for example get a `[4]` as a result and do a `.resize(3, 0)`, we now have `[4, 0, 0]`, breaking the return value as we use big endian for bytes

**Review by TomasArrachea — APPROVED** [review]

**Pull request merged**